cli/internal/state/state.go

```diff
@@ -7,23 +7,126 @@ SPDX-License-Identifier: AGPL-3.0-only
 // package state defines the structure of the Constellation state file.
 package state
 
+import (
+	"fmt"
+
+	"dario.cat/mergo"
+	"github.com/edgelesssys/constellation/v2/cli/internal/clusterid"
+	"github.com/edgelesssys/constellation/v2/internal/config"
+	"github.com/edgelesssys/constellation/v2/internal/file"
+)
+
 const (
 	// Version1 is the first version of the state file.
 	Version1 = "v1"
 )
 
+// ReadFromFile reads the state file at the given path and returns the state.
+func ReadFromFile(fileHandler file.Handler, path string) (*State, error) {
+	state := &State{}
+	if err := fileHandler.ReadYAML(path, &state); err != nil {
```

Comment:
just curious why you didn't choose
```
var state State
ReadYAML(path,&state)
return &state
```
why is a double pointer necessary in your implementation?

Reply:
This is just a typo actually. Thanks for pointing it out!